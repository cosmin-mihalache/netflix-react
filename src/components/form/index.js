import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
