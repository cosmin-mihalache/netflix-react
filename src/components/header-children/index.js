import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
