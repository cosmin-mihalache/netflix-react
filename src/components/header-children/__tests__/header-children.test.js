import React from 'react';
import { render } from '@testing-library/react';
import { HeaderChildren } from '../../../components';

describe('HeaderChildren', () => {
  it('renders the <HeaderChildren /> with populated data', () => {
    const { getByText } = render(
      <HeaderChildren>
        <HeaderChildren.Title>
          100% entertainment. EUR0.99 for your first 30 days.
        </HeaderChildren.Title>
        <HeaderChildren.SubTitle>
          Watch anywhere. Cancel anytime.
        </HeaderChildren.SubTitle>
      </HeaderChildren>
    );
    expect(
      getByText('100% entertainment. EUR0.99 for your first 30 days.')
    ).toBeTruthy();
    expect(getByText('Watch anywhere. Cancel anytime.')).toBeTruthy();
  });

  it('render the <HeaderChildren /> just a title', () => {
    const { queryByText, getByText } = render(
      <HeaderChildren>
        <HeaderChildren.Title>
          100% entertainment. EUR0.99 for your first 30 days.
        </HeaderChildren.Title>
        );
      </HeaderChildren>
    );
    expect(
      getByText('100% entertainment. EUR0.99 for your first 30 days.')
    ).toBeTruthy();
    expect(queryByText('Watch anywhere. Cancel anytime.')).toBeFalsy();
  });

  it('render the <HeaderChildren /> just a subTitle', () => {
    const { queryByText, getByText } = render(
      <HeaderChildren>
        <HeaderChildren.SubTitle>
          Watch anywhere. Cancel anytime.
        </HeaderChildren.SubTitle>
      </HeaderChildren>
    );
    expect(getByText('Watch anywhere. Cancel anytime.')).toBeTruthy();
    expect(
      queryByText('100% entertainment. EUR0.99 for your first 30 days.')
    ).toBeFalsy();
  });
});
