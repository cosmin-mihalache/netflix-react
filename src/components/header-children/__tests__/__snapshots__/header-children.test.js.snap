// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderChildren render the <HeaderChildren /> just a subTitle 1`] = `
<div
  class="header-children__Container-sc-1eq3yi1-0 eYPLAp"
>
  <h2
    class="header-children__SubTitle-sc-1eq3yi1-2 cRuqwl"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;

exports[`HeaderChildren render the <HeaderChildren /> just a title 1`] = `
<div
  class="header-children__Container-sc-1eq3yi1-0 eYPLAp"
>
  <h1
    class="header-children__Title-sc-1eq3yi1-1 eEvkuc"
  >
    100% entertainment. EUR0.99 for your first 30 days.
  </h1>
  );
</div>
`;

exports[`HeaderChildren renders the <HeaderChildren /> with populated data 1`] = `
<div
  class="header-children__Container-sc-1eq3yi1-0 eYPLAp"
>
  <h1
    class="header-children__Title-sc-1eq3yi1-1 eEvkuc"
  >
    100% entertainment. EUR0.99 for your first 30 days.
  </h1>
  <h2
    class="header-children__SubTitle-sc-1eq3yi1-2 cRuqwl"
  >
    Watch anywhere. Cancel anytime.
  </h2>
</div>
`;
